### PR TITLE
An optional `values` parameter in the `exec` method

### PR DESCRIPTION
**Pull request by slvrtrn**
## Summary

The `exec` method now accepts an optional `values` parameter, which allows you to pass the request body as a `Stream.Readable`. This can be useful in case of custom insert streaming with arbitrary ClickHouse data formats (which might not be explicitly supported and allowed by the client in the `insert` method yet). 

A very minimal example with CSV (from the tests, it does not make sense as CSV is supported via the normal `insert`, just an illustration):

```ts
const stream = Stream.Readable.from(['42,foobar,"[1,2]"'], {
  objectMode: false,
})
const execResult = await client.exec({
  query: `INSERT INTO ${tableName} FORMAT CSV`,
  values: stream,
})
// the result stream contains nothing useful for an insert 
// and should be immediately drained to release the socket
await drainStream(execResult.stream)

const rs = await client.query({
  query: `SELECT * FROM ${tableName}`,
  format: 'JSONEachRow',
})
expect(await rs.json()).toEqual([
  {
    id: '42',
    name: 'foobar',
    sku: [1, 2],
  },
])
```

## Checklist
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: packages/client-common/src/client.ts [hunk 1]
```diff
@@ -214,10 +226,14 @@ export class ClickHouseClient<Stream = unknown> {
    * but format clause is not applicable. The caller of this method is expected to consume the stream,
    * otherwise, the request will eventually be timed out.
    */
-  async exec(params: ExecParams): Promise<ExecResult<Stream>> {
+  async exec(
+    params: ExecParams | ExecParamsWithValues<Stream>,
```
Comment (slvrtrn) [inline packages/client-common/src/client.ts:230]: I avoided parametrizing ExecParams in this case, as it's a bit tricky to maintain full backward compatibility.

File: packages/client-node/src/connection/node_base_connection.ts [hunk 2]
```diff
@@ -368,20 +369,23 @@ export abstract class NodeBaseConnection
...
+    }
+    const searchParams = toSearchParams(toSearchParamsOptions)
     const { controller, controllerCleanup } = this.getAbortController(params)
     try {
       const { stream, summary, response_headers } = await this.request(
         {
           method: 'POST',
           url: transformUrl({ url: this.params.url, searchParams }),
-          body: params.query,
+          body: sendQueryInParams ? params.values : params.query,
```
Comment (mshustov) [inline packages/client-node/src/connection/node_base_connection.ts:388]: nit: I'd prefer to keep a single logic path to send a request, but let's postpone until the next major release when we can *always* send `query` in URL and `values` as body (even if it's empty)

File: packages/client-node/__tests__/integration/node_exec.test.ts [hunk 3]
```diff
@@ -66,4 +68,33 @@ describe('[Node.js] exec result streaming', () => {
       expect(await rs.json()).toEqual([{ number: '0' }])
     })
   })
+
+  describe('custom insert streaming with exec', () => {
```
Comment (mshustov) [inline packages/client-node/__tests__/integration/node_exec.test.ts:72]: can we add a test that it doesn't blow with an empty stream, closed stream, and other edge cases?

**Review by mshustov — APPROVED** [review]
